optuna_dashboard/artifact/_backend.py

```diff
@@ -123,8 +127,17 @@ def delete_artifact(study_id: int, trial_id: int, artifact_id: str) -> dict[str,
             return {"reason": "Cannot access to the artifacts."}
         artifact_store.remove(artifact_id)
 
-        attr_key = _artifact_prefix(trial_id) + artifact_id
-        storage.set_study_system_attr(study_id, attr_key, json.dumps(None))
+        # The metadata of the artifact is stored in one of the following three locations:
+        storage.set_study_system_attr(
+            study_id, _artifact_prefix(trial_id) + artifact_id, json.dumps(None)
+        )
+        storage.set_trial_system_attr(
+            trial_id, DASHBOARD_ARTIFACTS_ATTR_PREFIX + artifact_id, json.dumps(None)
+        )
+        storage.set_trial_system_attr(
```

Comment:
[question] Should we check that `ARTIFACTS_ATTR_PREFIX + artifact_id` exists in set_trial_system_attr?

Reply:
This was already removed.